diff --git a/garaged/benchmarks/src/garage_benchmarks/experiments/algos/ddpg_garage_tf.py b/garaged/benchmarks/src/garage_benchmarks/experiments/algos/ddpg_garage_tf.py
index 764ca11..8f38c98 100644
--- a/garaged/benchmarks/src/garage_benchmarks/experiments/algos/ddpg_garage_tf.py
+++ b/garaged/benchmarks/src/garage_benchmarks/experiments/algos/ddpg_garage_tf.py
@@ -2,15 +2,15 @@
 import gym
 import tensorflow as tf
 
-from garage import wrap_experiment
-from garage.envs import GarageEnv, normalize
-from garage.experiment import deterministic
-from garage.experiment import LocalTFRunner
-from garage.np.exploration_policies import AddOrnsteinUhlenbeckNoise
-from garage.replay_buffer import PathBuffer
-from garage.tf.algos import DDPG
-from garage.tf.policies import ContinuousMLPPolicy
-from garage.tf.q_functions import ContinuousMLPQFunction
+from garagei import wrap_experiment
+from garagei.envs import GarageEnv, normalize
+from garagei.experiment import deterministic
+from garagei.experiment import LocalTFRunner
+from garagei.np.exploration_policies import AddOrnsteinUhlenbeckNoise
+from garagei.replay_buffer import PathBuffer
+from garagei.tf.algos import DDPG
+from garagei.tf.policies import ContinuousMLPPolicy
+from garagei.tf.q_functions import ContinuousMLPQFunction
 
 hyper_parameters = {
     'policy_lr': 1e-4,
diff --git a/garaged/benchmarks/src/garage_benchmarks/experiments/algos/her_garage_tf.py b/garaged/benchmarks/src/garage_benchmarks/experiments/algos/her_garage_tf.py
index f9c47b5..db1b964 100644
--- a/garaged/benchmarks/src/garage_benchmarks/experiments/algos/her_garage_tf.py
+++ b/garaged/benchmarks/src/garage_benchmarks/experiments/algos/her_garage_tf.py
@@ -2,15 +2,15 @@
 import gym
 import tensorflow as tf
 
-from garage import wrap_experiment
-from garage.envs import GarageEnv, normalize
-from garage.experiment import deterministic
-from garage.experiment import LocalTFRunner
-from garage.np.exploration_policies import AddOrnsteinUhlenbeckNoise
-from garage.replay_buffer import HERReplayBuffer
-from garage.tf.algos import DDPG
-from garage.tf.policies import ContinuousMLPPolicy
-from garage.tf.q_functions import ContinuousMLPQFunction
+from garagei import wrap_experiment
+from garagei.envs import GarageEnv, normalize
+from garagei.experiment import deterministic
+from garagei.experiment import LocalTFRunner
+from garagei.np.exploration_policies import AddOrnsteinUhlenbeckNoise
+from garagei.replay_buffer import HERReplayBuffer
+from garagei.tf.algos import DDPG
+from garagei.tf.policies import ContinuousMLPPolicy
+from garagei.tf.q_functions import ContinuousMLPQFunction
 
 hyper_parameters = {
     'policy_lr': 1e-3,
diff --git a/garaged/benchmarks/src/garage_benchmarks/experiments/algos/ppo_garage_pytorch.py b/garaged/benchmarks/src/garage_benchmarks/experiments/algos/ppo_garage_pytorch.py
index e0533cd..fdafb57 100644
--- a/garaged/benchmarks/src/garage_benchmarks/experiments/algos/ppo_garage_pytorch.py
+++ b/garaged/benchmarks/src/garage_benchmarks/experiments/algos/ppo_garage_pytorch.py
@@ -2,13 +2,13 @@
 import gym
 import torch
 
-from garage import wrap_experiment
-from garage.envs import GarageEnv, normalize
-from garage.experiment import deterministic, LocalRunner
-from garage.torch.algos import PPO as PyTorch_PPO
-from garage.torch.optimizers import OptimizerWrapper
-from garage.torch.policies import GaussianMLPPolicy as PyTorch_GMP
-from garage.torch.value_functions import GaussianMLPValueFunction
+from garagei import wrap_experiment
+from garagei.envs import GarageEnv, normalize
+from garagei.experiment import deterministic, LocalRunner
+from garagei.torch.algos import PPO as PyTorch_PPO
+from garagei.torch.optimizers import OptimizerWrapper
+from garagei.torch.policies import GaussianMLPPolicy as PyTorch_GMP
+from garagei.torch.value_functions import GaussianMLPValueFunction
 
 hyper_parameters = {
     'n_epochs': 500,
diff --git a/garaged/examples/tf/rl2_ppo_metaworld_ml1_push.py b/garaged/examples/tf/rl2_ppo_metaworld_ml1_push.py
index bffff4d..3cdcce4 100644
--- a/garaged/examples/tf/rl2_ppo_metaworld_ml1_push.py
+++ b/garaged/examples/tf/rl2_ppo_metaworld_ml1_push.py
@@ -4,16 +4,16 @@
 import click
 import metaworld.benchmarks as mwb
 
-from garage import wrap_experiment
-from garage.experiment import LocalTFRunner
-from garage.experiment import task_sampler
-from garage.experiment.deterministic import set_seed
-from garage.np.baselines import LinearFeatureBaseline
-from garage.sampler import LocalSampler
-from garage.tf.algos import RL2PPO
-from garage.tf.algos.rl2 import RL2Env
-from garage.tf.algos.rl2 import RL2Worker
-from garage.tf.policies import GaussianGRUPolicy
+from garagei import wrap_experiment
+from garagei.experiment import LocalTFRunner
+from garagei.experiment import task_sampler
+from garagei.experiment.deterministic import set_seed
+from garagei.np.baselines import LinearFeatureBaseline
+from garagei.sampler import LocalSampler
+from garagei.tf.algos import RL2PPO
+from garagei.tf.algos.rl2 import RL2Env
+from garagei.tf.algos.rl2 import RL2Worker
+from garagei.tf.policies import GaussianGRUPolicy
 
 
 @click.command()
diff --git a/garaged/setup.py b/garaged/setup.py
index 512879b..7a1bdba 100644
--- a/garaged/setup.py
+++ b/garaged/setup.py
@@ -12,15 +12,19 @@ REQUIRED = [
     # Please keep alphabetized
     'akro',
     'click>=2.0',
-    'cloudpickle==1.3',
+    # 'cloudpickle==1.3',
+    'cloudpickle',
     'cma==2.7.0',
     #'dowel==0.0.3',
-    'gym[atari,box2d,classic_control]' + GYM_VERSION,
-    'joblib<0.13,>=0.12',
+    #'gym[atari,box2d,classic_control]' + GYM_VERSION,
+    'gym[box2d,classic_control]' + GYM_VERSION, # MOVED TO ALE-PY
+    #'joblib<0.13,>=0.12',
+    'joblib',
     'numpy>=1.14.5',
     'psutil',
     'python-dateutil',
-    'ray<=0.8.6',
+    #'ray<=0.8.6',
+    'ray',
     'scikit-image',
     'scipy',
     'setproctitle>=1.0',
diff --git a/iod/dads.py b/iod/dads.py
index e7fd43c..e7f04d5 100644
--- a/iod/dads.py
+++ b/iod/dads.py
@@ -1,10 +1,10 @@
 import numpy as np
 import torch
 
-from iod.metra import METRA_bl
+from iod.metra import METRA
 
 
-class DADS(METRA_bl):
+class DADS(METRA):
     def __init__(
             self,
             **kwargs,
diff --git a/iod/iod.py b/iod/iod.py
index d0c5774..0ee5a2f 100644
--- a/iod/iod.py
+++ b/iod/iod.py
@@ -20,7 +20,7 @@ import random
 from envs.AntMazeEnv import MazeWrapper, GoalReachingMaze, plot_trajectories, plot_value
 import matplotlib.pyplot as plt
 import os
-from iod.ant_eval import *
+# from iod.ant_eval import *
 from iod.agent import *
 from iod.viz_utils import PlotMazeTrajDist, PlotMazeTrajWindowDist, viz_dist_circle, PlotGMM
 
diff --git a/requirements.txt b/requirements.txt
index abdc87c..00bc348 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -41,7 +41,8 @@ executing==1.2.0
 fasteners==0.18
 fastjsonschema==2.17.1
 filelock==3.12.0
-flatbuffers==23.5.26
+# flatbuffers==23.5.26
+flatbuffers>=24.3.25
 fonttools==4.39.4
 fqdn==1.5.1
 future==0.18.3
@@ -50,18 +51,20 @@ gitdb==4.0.10
 GitPython==3.1.31
 glfw==2.5.9
 google-auth==2.19.1
-google-auth-oauthlib==0.4.6
+# google-auth-oauthlib==0.4.6
+google-auth-oauthlib
 google-pasta==0.2.0
 grpcio==1.54.2
 gym==0.17.2
 gym-notices==0.0.8
-h5py==3.8.0
+# h5py==3.8.0
+h5py>=3.10.0
 idna==3.4
 imageio==2.31.0
 imageio-ffmpeg==0.4.8
 importlib-metadata==6.6.0
 importlib-resources==5.12.0
-install==1.3.5
+#install==1.3.5
 ipykernel==6.23.1
 ipython==8.12.2
 ipython-genutils==0.2.0
@@ -85,7 +88,8 @@ jupyterlab==3.6.1
 jupyterlab-pygments==0.2.2
 jupyterlab-vim==0.16.0
 jupyterlab_server==2.22.1
-keras==2.10.0
+# keras==2.10.0
+keras==2.15.0
 Keras-Preprocessing==1.1.2
 kiwisolver==1.4.4
 labmaze==1.0.6
@@ -128,7 +132,9 @@ platformdirs==3.5.1
 proglog==0.1.10
 prometheus-client==0.17.0
 prompt-toolkit==3.0.38
-protobuf==3.19.4
+# protobuf==3.19.4
+# protobuf>=3.20.1
+protobuf>=3.20.3,<5
 psutil==5.9.5
 ptyprocess==0.7.0
 pure-eval==0.2.2
@@ -166,12 +172,17 @@ soupsieve==2.4.1
 stack-data==0.6.2
 sympy==1.12
 tabulate==0.9.0
-tensorboard==2.10.1
-tensorboard-data-server==0.6.1
+# tensorboard==2.10.1ad
+tensorboard>=2.11
+# tensorboard-data-server==0.6.1
+tensorboard-data-server
 tensorboard-plugin-wit==1.8.1
 tensorboardX==2.6
-tensorflow-estimator==2.10.0
-tensorflow==2.10.0
+# tensorflow-estimator==2.10.0
+tensorflow-estimator >= 2.15.0, < 2.16
+# tensorflow==2.10.0
+# tensorflow>=2.11
+tensorflow==2.15.0
 termcolor==2.3.0
 terminado==0.17.1
 threadpoolctl==3.1.0
@@ -192,7 +203,8 @@ webcolors==1.13
 webencodings==0.5.1
 websocket-client==1.5.2
 Werkzeug==2.3.4
-wrapt==1.15.0
+# wrapt==1.15.0
+wrapt >= 1.11.0, < 1.15
 y-py==0.5.9
 ypy-websocket==0.8.2
 zipp==3.15.0
diff --git a/tests/main.py b/tests/main.py
index 0b26bc8..704b29f 100644
--- a/tests/main.py
+++ b/tests/main.py
@@ -1,4 +1,5 @@
 #!/usr/bin/env python3
+
 import tempfile
 import dowel_wrapper
 assert dowel_wrapper is not None
@@ -14,10 +15,13 @@ import torch.multiprocessing as mp
 if 'mac' in platform.platform():
     pass
 else:
+    print("HEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE")
     os.environ['MUJOCO_GL'] = 'egl'
     if 'SLURM_STEP_GPUS' in os.environ:
         os.environ['EGL_DEVICE_ID'] = os.environ['SLURM_STEP_GPUS']
 
+    os.environ["MUJOCO_PY_FORCE_CPU_RENDER"] = "true"
+
 # import better_exceptions
 import numpy as np
 
